tools/fds/server.cpp

```diff
@@ -561,6 +561,16 @@ int fastdds_discovery_server(
         // Handle signal SIGINT for every thread
         signal(SIGINT, sigint_handler);
         signal(SIGTERM, sigint_handler);
+        signal(SIGQUIT, sigint_handler);
+        signal(SIGILL, sigint_handler);
+        signal(SIGABRT, sigint_handler);
+        signal(SIGBUS, sigint_handler);
+        signal(SIGFPE, sigint_handler);
+#ifndef _WIN32
+        signal(SIGSEGV, sigint_handler);
+        signal(SIGPIPE, sigint_handler);
+        signal(SIGHUP, sigint_handler);
+#endif // ifndef _WIN32
```

Comment:
I would not capture some of these:

`SIGILL`, `SIGBUS`, `SIGABRT`, `SIGFPE`, and `SIGSEGV` are usually triggered due to internal logic errors, and they produce a core file that would allow us to further investigate the issue.

We are not using pipes, so `SIGPIPE` doesn't make sense either

```suggestion
        signal(SIGQUIT, sigint_handler);
#ifndef _WIN32
        signal(SIGHUP, sigint_handler);
#endif // ifndef _WIN32
```